napari/_qt/layers/qt_base_layer.py

```diff
@@ -47,62 +62,38 @@ def __init__(self, layer):
         sld.setMinimum(0)
         sld.setMaximum(100)
         sld.setSingleStep(1)
-        sld.valueChanged.connect(self.changeOpacity)
+        sld.valueChanged.connect(lambda v: self.events.opacity(v / 100))
         self.opacitySlider = sld
-        self._on_opacity_change()
 
         blend_comboBox = QComboBox(self)
         blend_comboBox.addItems(Blending.keys())
-        index = blend_comboBox.findText(
-            self.layer.blending, Qt.MatchFixedString
-        )
-        blend_comboBox.setCurrentIndex(index)
-        blend_comboBox.activated[str].connect(self.changeBlending)
+        blend_comboBox.activated[str].connect(self.events.blending)
         self.blendComboBox = blend_comboBox
 
-    def changeOpacity(self, value):
-        """Change opacity value on the layer model.
-
-        Parameters
-        ----------
-        value : float
-            Opacity value for shapes.
-            Input range 0 - 100 (transparent to fully opaque).
-        """
-        with self.layer.events.blocker(self._on_opacity_change):
-            self.layer.opacity = value / 100
+        # Once EVH refactor is done, these can be moved to an initialization
+        # outside of this object
+        self._on_opacity_change(self.layer.opacity)
+        self._on_blending_change(self.layer.blending)
```

Comment:
Yeah, this is a good question. It's hard right now as we don't have an explicit concept of a controller which is I guess where these imports, initialization and wiring up would be done (as it would be the one place which knows about both model and view). We can't do it in the event handler itself as we import the event handler into the layer model directly so that we can edit the layer model without the views, so we need an additional concept here. I'll work on some architecture docs in a new PR and we can revisit this there.

As to the grouping, I was grouping the soon to be obsolete properties together to make easier to remove them later.

Reply:
Yeah, this is a good question. It's hard right now as we don't have an explicit concept of a controller which is I guess where these imports, initialization and wiring up would be done (as it would be the one place which knows about both model and view). We can't do it in the event handler itself as we import the event handler into the layer model directly so that we can edit the layer model without the views, so we need an additional concept here. I'll work on some architecture docs in a new PR and we can revisit this there.

As to the grouping, I was grouping the soon to be obsolete properties together to make easier to remove them later.

Reply:
Ooo - ok, I have one idea - we could add it to the `register_listener` method of the event handler. It would be nice if we had a standardized way of figuring out what the event values should be and then re-emitting them.  i.e. when you register the `qt_image_layer` we see that it has the `_on_visible_change` method and we know that the `Image` layer emits the `visible` event, and so we call the current `visible` value lives at `layer.visible` so we call `_on_visible_change(layer.visible)`?

We might be baking in a lot of assumptions there, but it would remove the need to a lot of initialization code throughout the repo. Curious @tlambert03 @jni @ziyangczi what you think of this idea?